### DISCUSSION TITLE
How can I run NixOS on WSL2 on Windows 11 inside VirtualBox?

### DISCUSSION THREAD

**Discussion by madorian** [nix-community/NixOS-WSL#403] [Q&A]
I assume you develop NixOS-WSL on NixOS itself with Windows in a VirtualBox or similar?

The reason I ask is because I can't get WSL2 to work on VirtualBox. 

My understanding is that I need a newer VirtualBox which supports a more complete nested hypervisor?

**Comment by SuperSandro2000** [comment]
At least for me, I am running NixOS-WSL on Windows 10/11 on bare metal.

WSL 2 requires HyperV and with that nested virtualisation.

**Comment by nzbr** [comment]
From the [VirtualBox documentation](https://docs.oracle.com/en/virtualization/virtualbox/6.0/admin/nested-virt.html):
> From the VirtualBox Manager, select the Enable Nested VT-x/AMD-V check box on the Processor tab. To disable the feature, deselect the check box. 

Once you have that enabled, you should be able to setup WSL inside your Windows VM. Once you have that, you should be able to install NixOS the same way you would on a bare-metal Windows install

**Reply by madorian** [reply]
Have you confirmed this yourself? Cause according to VirtualBox Dev, it's not possible before you install a development snapshot of VirtualBox and even that is flaky: https://forums.virtualbox.org/viewtopic.php?t=110039

**Reply by nzbr** [reply]
I can't easily test this, because I already use WSL2 on my Windows Host, and therefore can't enable nested virtualization
